File: c/hoppity.c
#include <stdio.h>
#include <stdlib.h>

void usage(char* name)
{
  printf("Usage: %s <file>\n", name);
  exit(1);
}

void hoppity(int n)
{
  for (int i = 1; i <= n; i++)
  {
      char* s = i % 15 == 0 ? "Hop" \
              : i %  5 == 0 ? "Hophop" \
              : i %  3 == 0 ? "Hoppity" \
              : NULL;
      if (s != NULL)
      {
        printf("%s\n", s);
      }
  }
}

int main(int argc, char* argv[])
{
  if (argc != 2)
  {
    usage(argv[0]);
  }

  freopen(argv[1], "r", stdin);

  int n;
  scanf("%d", &n);

  hoppity(n);

  return 0;
}

File: java/hoppity.java
import java.io.File;
import java.util.Scanner;

public class hoppity
{
  private static void usage()
  {
    System.err.println("Usage: java " + hoppity.class.getName() + " <file>");
    System.exit(1);
  }

  private static void hoppity(final int n)
  {
    for (int i = 1; i <= n; i++)
    {
      final String s = i % 15 == 0 ? "Hop"
                     : i %  5 == 0 ? "Hophop"
                     : i %  3 == 0 ? "Hoppity"
                     : null;
      if (s != null)
      {
        System.out.println(s);
      }
    }
  }

  public static void main(final String[] args) throws Exception
  {
    if (args.length != 1 || args[0].trim().length() == 0)
    {
      usage();
    }

    final Scanner in = new Scanner(new File(args[0]));
    final int n = in.nextInt();
    hoppity(n);
  }
}
